internal/scheduler/context/context.go

```diff
@@ -116,26 +117,51 @@ func (sctx *SchedulingContext) AddQueueSchedulingContext(queue string, priorityF
 }
 
 func (sctx *SchedulingContext) String() string {
+	return sctx.ReportString(0)
+}
+
+func (sctx *SchedulingContext) ReportString(verbosity int32) string {
 	var sb strings.Builder
 	w := tabwriter.NewWriter(&sb, 1, 1, 1, ' ', 0)
 	fmt.Fprintf(w, "Started:\t%s\n", sctx.Started)
 	fmt.Fprintf(w, "Finished:\t%s\n", sctx.Finished)
 	fmt.Fprintf(w, "Duration:\t%s\n", sctx.Finished.Sub(sctx.Started))
-	fmt.Fprintf(w, "Total capacity:\t%s\n", sctx.TotalResources.CompactString())
-	fmt.Fprintf(w, "Jobs scheduled:\t%d\n", sctx.NumScheduledJobs)
-	fmt.Fprintf(w, "Total scheduled resources:\t%s\n", sctx.ScheduledResources.CompactString())
-	fmt.Fprintf(
-		w, "Scheduled queues:\t%v\n",
-		maps.Keys(
-			armadamaps.Filter(
-				sctx.QueueSchedulingContexts,
-				func(_ string, qctx *QueueSchedulingContext) bool {
-					return len(qctx.SuccessfulJobSchedulingContexts) > 0
-				},
-			),
-		),
-	)
 	fmt.Fprintf(w, "Termination reason:\t%s\n", sctx.TerminationReason)
+	fmt.Fprintf(w, "Available resources before scheduling:\t%s\n", sctx.TotalResources.CompactString())
+	fmt.Fprintf(w, "Scheduled resources:\t%s\n", sctx.ScheduledResources.CompactString())
+	fmt.Fprintf(w, "Number of jobs scheduled:\t%d\n", sctx.NumScheduledJobs)
+	if verbosity <= 0 {
+		fmt.Fprintf(
+			w,
+			"Scheduled queues:\t%v\n",
+			maps.Keys(
+				armadamaps.Filter(
+					sctx.QueueSchedulingContexts,
+					func(_ string, qctx *QueueSchedulingContext) bool {
+						return len(qctx.SuccessfulJobSchedulingContexts) > 0
+					},
+				),
+			),
+		)
+		fmt.Fprintf(
+			w,
+			"Preempted queues:\t%v\n",
+			maps.Keys(
+				armadamaps.Filter(
+					sctx.QueueSchedulingContexts,
+					func(_ string, qctx *QueueSchedulingContext) bool {
+						return len(qctx.EvictedJobsById) > 0
+					},
+				),
+			),
+		)
+	} else {
+		fmt.Fprint(w, "Queues:\n")
+		for queueName, qctx := range sctx.QueueSchedulingContexts {
+			fmt.Fprintf(w, "\t%s:\n", queueName)
+			fmt.Fprintf(w, indent.String("\t\t", qctx.ReportString(verbosity-1)))
```

Comment:
I thought that this idea of decreasing the verbosity level by one for nested calls to `ReportString` was kind of cute.

This would probably be easier to follow if we used a smaller number of verbosity levels (for example: "compact"/"detailed"/"verbose" corresponding to the current behavior of `0`/`1`/`2`).